target_chains/ethereum/contracts/contracts/pyth/Pyth.sol

```diff
@@ -101,7 +102,30 @@ abstract contract Pyth is
     function getUpdateFee(
         bytes[] calldata updateData
     ) public view override returns (uint feeAmount) {
-        return singleUpdateFeeInWei() * updateData.length;
+        uint8 totalNumUpdates = 0;
+        for (uint i = 0; i < updateData.length; i++) {
+            if (
+                updateData[i].length > 4 &&
+                UnsafeBytesLib.toUint32(updateData[i], 0) == ACCUMULATOR_MAGIC
+            ) {
+                (
+                    uint offset,
+                    UpdateType updateType
+                ) = extractUpdateTypeFromAccumulatorHeader(updateData[i]);
+                if (updateType != UpdateType.WormholeMerkle) {
+                    revert PythErrors.InvalidUpdateData();
+                }
+                totalNumUpdates += parseWormholeMerkleHeaderNumUpdates(
+                    updateData[i],
+                    offset
+                );
+            }
+        }
+        if (totalNumUpdates > 0) {
+            return totalNumUpdates * singleUpdateFeeInWei();
+        } else {
+            return updateData.length * singleUpdateFeeInWei();
+        }
```

Comment:
You can make an else statement above and add totalNumUpdates by 1 if it's not accumulator. It will be cleaner.